langlib/lang.object/src/main/ballerina/object.bal

```diff
@@ -13,3 +13,56 @@
 // KIND, either express or implied.  See the License for the
 // specific language governing permissions and limitations
 // under the License.
+
+# Distinct Iterable type.
+# An object can make itself iterable by using `*object:Iterable;`,
+# and then defining an `iterator` method.
+public type Iterable distinct object {
+    # Create a new iterator.
+    #
+    # + return - a new iterator object
+    public function iterator() returns object {
+        public isolated function next() returns record {| any|error value; |}|error?;
+    };
+};
+
+# Integer range expression is represented using `IntRange` object.
+#
+# + iStart - start expression of range expression
+# + iEnd - second expression on range expression
+# + iCurrent - current cursor
+public class IntRange {
```

Comment:
Didn't we decide to have this in the internal package?

Reply:
Need to discuss this. According to spec **Range expression** should be available to be used anywhere and not just in `foreach`. Therefore users should be able to assign range expression to a variable. If I hide this in the internal package we have a type for integer range but it is a hidden type. 
Basically, code action creates a variable for integer range will create type defined in the internal package.